syncer/checkpoint.go

```diff
@@ -485,15 +490,20 @@ func (cp *RemoteCheckPoint) createTable() error {
 			UNIQUE KEY uk_id_schema_table (id, cp_schema, cp_table)
 		)`, tableName)
 	args := make([]interface{}, 0)
-	_, err := cp.db.executeSQL(cp.tctx, []string{sql2}, [][]interface{}{args}...)
+	_, err := cp.dbConn.executeSQL(cp.tctx, []string{sql2}, [][]interface{}{args}...)
 	cp.tctx.L().Info("create checkpoint table", zap.String("statement", sql2))
 	return err
 }
 
 // Load implements CheckPoint.Load
 func (cp *RemoteCheckPoint) Load() error {
 	query := fmt.Sprintf("SELECT `cp_schema`, `cp_table`, `binlog_name`, `binlog_pos`, `is_global` FROM `%s`.`%s` WHERE `id`='%s'", cp.schema, cp.table, cp.id)
-	rows, err := cp.db.querySQL(cp.tctx, query)
+	rows, err := cp.dbConn.querySQL(cp.tctx, query)
+	defer func() {
+		if rows != nil {
+			rows.Close()
```

Comment:
why do you move `rows.Close()` to here?

Reply:
can we move `LoadCheckpointFailed` before `rows, err := cp.dbConn.querySQL(cp.tctx, query)`?

Reply:
if we move `LoadCheckpointFailed` before `rows, err := cp.dbConn.querySQL(cp.tctx, query)`, the `err` in `LoadCheckpointFailed` will be overwrite.

Reply:
Is that a problem? with the code like
```golang
var err error
failpoint.Inject("LoadCheckpointFailed", func(val failpoint.Value) {
    err = tmysql.NewErr(uint16(val.(int)))
    log.L().Warn("Load failed", zap.String("failpoint", "LoadCheckpointFailed"), zap.Error(err))
})

if err != nil {
    return terror.WithScope(err, terror.ScopeDownstream)
}

query := fmt.Sprintf("SELECT `cp_schema`, `cp_table`, `binlog_name`, `binlog_pos`, `is_global` FROM `%s`.`%s` WHERE `id`='%s'", cp.schema, cp.table, cp.id)
rows, err := cp.dbConn.querySQL(cp.tctx, query)
defer func() {
    if rows != nil {
        rows.Close()
    }
}()

if err != nil {
    return terror.WithScope(err, terror.ScopeDownstream)
}
```

Reply:
the code has no problem, but the purpose of `LoadCheckpointFailed` here should be we may met same error in  `cp.dbConn.querySQL`? if so, why not put `LoadCheckpointFailed` after `cp.dbConn.querySQL`, I think it's more clear

Reply:
`LoadCheckpointFailed` should be meeting some error when loading (this `Load` method) checkpoint?